requests/auth.py

```diff
@@ -27,6 +27,12 @@
 
 def _basic_auth_str(username, password):
     """Returns a Basic Auth string."""
+    
+    if isinstance(username, bytes):
+        username = username.decode('latin1')
+
+    if isinstance(password, bytes):
+        password = password.decode('latin1')
```

Comment:
Let's not do it this way around. Given that we need to re-encode immediately (`b64encode` only works on `bytes` objects), we should instead check for *unicode* strings and encode them using "latin1", then change the `authstr` line to do `b"%s:%s" % (username, password)` where `username` and `password` are both `bytes`.